tgui/packages/tgui/interfaces/ClothingBooth/type.ts

```diff
@@ -1,21 +1,72 @@
+import { BooleanLike } from "common/react";
+
 export interface ClothingBoothData {
-  clothingBoothCategories: ClothingBoothCategory[];
-  money: number;
+  accountBalance?: number;
+  cash?: number;
+  catalogue: Record<string, ClothingBoothGroupingData>;
+  everythingIsFree: boolean;
   name: string;
   previewHeight: number;
   previewIcon: string;
-  selectedItemCost: number;
-  selectedItemName: string;
+  previewShowClothing: BooleanLike;
+  scannedID?: string;
+  selectedGroupingName: string | null;
+  selectedItemName: string | null;
+  tags: Record<string, ClothingBoothGroupingTagsData>;
 }
 
-export interface ClothingBoothCategory {
-  category: string;
-  items: CategoryItems[];
+export interface ClothingBoothGroupingData {
+  clothingbooth_items: Record<string, ClothingBoothItemData>;
+  cost_max: number;
+  cost_min: number;
+  grouping_tags: string[];
+  list_icon: string;
+  name: string;
+  slot: ClothingBoothSlotKey;
 }
 
-export interface CategoryItems {
+export interface ClothingBoothItemData {
+  name: string;
   cost: number;
-  img: string;
+  swatch_background_colour?: string;
+	swatch_foreground_shape?: string;
+	swatch_foreground_colour?: string;
+}
+
+export interface SwatchForegroundProps {
+  colour: string;
```

Comment:
Not sure why you'd want this to be a single color, in that a single color could just be programmatically applied as a fill style to an SVG instead of having to do this way.

My understanding of intent for these icons was that they could support multiple shapes/colors at once? Which would necessitate a `string[]`.

If you only want these to have a single color, then these can just be `.svg` files as before. Set the `fill` to be the color you want.

Also, `colour` -> `color`. Mixing British in to web stuff is dangerous.

Reply:
You're right, they should only be a single colour, I was just sent down this wormhole trying to figure out `.svg` styling this entire time because of the restrictions of IE11. Importing vector images from `.svg` has caused an unneccessary amount of pain on my end.